tap_stripe/__init__.py

```diff
@@ -557,15 +561,38 @@ def sync_stream(stream_name):
     stream_field_whitelist = json.loads(Context.config.get('whitelist_map', '{}')).get(stream_name)
 
     extraction_time = singer.utils.now()
-    replication_key = metadata.get(stream_metadata, (), 'valid-replication-keys')[0]
+    should_sync_parent_stream = True
+
+    if Context.is_sub_stream(stream_name):
+        replication_key = STREAM_REPLICATION_KEY.get(PARENT_STREAM_MAP.get(stream_name))
+        sub_stream_name = stream_name
+        stream_name = PARENT_STREAM_MAP.get(stream_name)
+        should_sync_parent_stream = is_parent_selected(sub_stream_name)
+    else:
+        replication_key = metadata.get(stream_metadata, (), 'valid-replication-keys')[0]
+
     # Invoice Items bookmarks on `date`, but queries on `created`
     filter_key = 'created' if stream_name == 'invoice_items' else replication_key
 
+    sub_stream_name = SUB_STREAMS.get(stream_name)
+
     # Get bookmark for the stream
     stream_bookmark = get_bookmark_for_stream(stream_name, replication_key)
-
     bookmark = stream_bookmark
 
+    # If there is a sub-stream and its selected, get its bookmark (or the start date if no bookmark)
+    should_sync_sub_stream = sub_stream_name and Context.is_selected(sub_stream_name)
+
+    if should_sync_sub_stream and should_sync_parent_stream:
+        sub_stream_bookmark = get_bookmark_for_sub_stream(sub_stream_name)
+
+        # if there is a sub stream, set bookmark to sub stream's bookmark
+        # since we know it must be earlier than the stream's bookmark
+        if sub_stream_bookmark != stream_bookmark:
+            bookmark = min(stream_bookmark, sub_stream_bookmark)
```

Comment:
We can take this each time no need to check the condition, if both are similar then also min will give the same result.

Reply:
Can we change like below: 
```
    if should_sync_sub_stream:
        bookmark = get_bookmark_for_sub_stream(sub_stream_name)
    else:
        bookmark = None
```